Xamarin.Essentials/Calendar/Calendar.uwp.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Windows.ApplicationModel.Appointments;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Calendar
+    {
+        static async Task<IEnumerable<DeviceCalendar>> PlatformGetCalendarsAsync()
+        {
+            await Permissions.RequireAsync(PermissionType.CalendarRead);
+
+            var instance = await CalendarRequest.GetInstanceAsync();
+            var uwpCalendarList = await instance.FindAppointmentCalendarsAsync(FindAppointmentCalendarsOptions.IncludeHidden);
+            var calendars = new List<DeviceCalendar>();
+            foreach (var c in uwpCalendarList)
+            {
+                calendars.Add(new DeviceCalendar()
+                {
+                    Id = c.LocalId,
+                    Name = c.DisplayName
+                });
+            }
+            return calendars;
+        }
+
+        static async Task<IEnumerable<DeviceEvent>> PlatformGetEventsAsync(string calendarId = null, DateTimeOffset? startDate = null, DateTimeOffset? endDate = null)
+        {
+            await Permissions.RequireAsync(PermissionType.CalendarRead);
+
+            var options = new FindAppointmentsOptions();
+            options.FetchProperties.Add(AppointmentProperties.Subject);
+            options.FetchProperties.Add(AppointmentProperties.StartTime);
+            options.FetchProperties.Add(AppointmentProperties.Duration);
+            options.FetchProperties.Add(AppointmentProperties.AllDay);
+            var sDate = startDate ?? DateTimeOffset.Now.Add(defaultStartTimeFromNow);
+            var eDate = endDate ?? sDate.Add(defaultEndTimeFromStartTime);
+
+            if (eDate < sDate)
+                eDate = sDate;
+
+            var instance = await CalendarRequest.GetInstanceAsync();
+            var events = await instance.FindAppointmentsAsync(sDate, eDate.Subtract(sDate), options);
+            var eventList = new List<DeviceEvent>();
+            foreach (var e in events)
```

Comment:
linq - use where clause

Reply:
Addressed in:
https://github.com/builttoroam/Essentials/pull/43